### PR TITLE
Functional Tests : WS - Countries : CRUD

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Functional Tests : WS - Countries : CRUD
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: & Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/4875606125
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

File: tests/UI/campaigns/functional/WS/01_countriesCRUD.ts [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+        const pageTitle = await addCountryPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(addCountryPage.pageTitleEdit);
+      });
+
+      it('should check all values', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'checkEditCountryValues', baseContext);
+
+        const createDocument: Document = domParser.parseFromString(xmlCreate);
+
+        const xmlValueIDZone = xpath.select1('string(/prestashop/country/id_zone)', createDocument) as string;
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/WS/01_countriesCRUD.ts]: Maybe it's better to create a helper for ws methods, wdyth ?

File: tests/UI/pages/BO/international/locations/countries/add.ts [hunk 2]
```diff
@@ -98,6 +101,96 @@ class AddCountry extends BOBasePage {
 
     return this.getTextContent(page, this.alertBlock);
   }
+
+  /**
+   * Get the value of an input
+   *
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/international/locations/countries/add.ts]:
```suggestion
```

File: tests/UI/pages/BO/international/locations/countries/add.ts [hunk 3]
```diff
@@ -98,6 +101,96 @@ class AddCountry extends BOBasePage {
...
+      default:
+        throw new Error(`Field ${input} was not found`);
+    }
+
+    return page.inputValue(selector);
+  }
+
+  /**
+   * Get the value of an select
+   *
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/international/locations/countries/add.ts]:
```suggestion
  
```

File: tests/UI/pages/BO/international/locations/countries/add.ts [hunk 4]
```diff
@@ -98,6 +101,96 @@ class AddCountry extends BOBasePage {
...
+      default:
+        throw new Error(`Field ${input} was not found`);
+    }
+
+    return page.$eval(selector, (node: HTMLSelectElement) => node.value);
+  }
+
+  /**
+   * Get the value of an input
+   *
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/international/locations/countries/add.ts]:
```suggestion
```

File: tests/UI/pages/BO/advancedParameters/webservice/index.ts [hunk 5]
```diff
@@ -428,6 +440,19 @@ class WebService extends BOBasePage {
 
     return this.getPaginationLabel(page);
   }
+
+  /* Form methods */
+  /**
+   * Set the webservice status
+   * @param page
+   * @param status
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/advancedParameters/webservice/index.ts]:
```suggestion
   * @param status
   @return {Promise<string>}
```

File: tests/UI/pages/BO/international/locations/countries/add.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/pages/BO/international/locations/countries/add.ts]: Fixed

File: tests/UI/pages/BO/international/locations/countries/add.ts (hunk 3 above)
Comment (Progi1984) [reply tests/UI/pages/BO/international/locations/countries/add.ts]: Fixed

File: tests/UI/pages/BO/international/locations/countries/add.ts (hunk 4 above)
Comment (Progi1984) [reply tests/UI/pages/BO/international/locations/countries/add.ts]: Fixed

File: tests/UI/pages/BO/advancedParameters/webservice/index.ts (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/webservice/index.ts]: Fixed

File: tests/UI/campaigns/functional/WS/01_countriesCRUD.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/WS/01_countriesCRUD.ts]: Fixed

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**